### PR TITLE
Add CopyImage IMAGE_TYPE VU

### PR DESCRIPTION
**Pull request by spencer-lunarg**
Adds `VUID-vkCmdCopyImage-srcImage-07742`, `VUID-VkCopyImageInfo2-srcImage-07742`,  `VUID-vkCmdCopyImage-srcImage-07743` and `VUID-VkCopyImageInfo2-srcImage-07743`

### CODE REVIEW

File: layers/buffer_validation.cpp [hunk 1]
```diff
@@ -3134,6 +3134,8 @@ bool CoreChecks::ValidateCmdCopyImage(VkCommandBuffer commandBuffer, VkImage src
     const CMD_BUFFER_STATE &cb_state = *cb_state_ptr;
     const VkFormat src_format = src_image_state->createInfo.format;
     const VkFormat dst_format = dst_image_state->createInfo.format;
+    const VkImageType src_image_type = src_image_state->createInfo.imageType;
+    const VkImageType dst_image_type = dst_image_state->createInfo.imageType;
```
Comment (juan-lunarg) [inline layers/buffer_validation.cpp:3138]:
```suggestion
    const VkImageType dst_image_type = dst_image_state->createInfo.imageType;
    const bool srcIs3D = (VK_IMAGE_TYPE_3D == src_image_type);
    const bool dstIs3D =  (VK_IMAGE_TYPE_3D == dst_image_type);
```

**Review by juan-lunarg — COMMENTED** [review]
Avoided repeated 2d/3d checks by creating bools before the loop.

File: tests/vklayertests_command.cpp [hunk 2]
```diff
@@ -2447,18 +2447,13 @@ TEST_F(VkLayerTest, MiscImageLayerTests) {
 
 TEST_F(VkLayerTest, CopyImageTypeExtentMismatch) {
     // Image copy tests where format type and extents don't match
```
Comment (juan-lunarg) [inline tests/vklayertests_command.cpp:2449]:
```suggestion
    TEST_DESCRIPTION("Image copy tests where format type and extents don't match");
```

**Review by juan-lunarg — APPROVED** [review]
Minor nitpick but looks good :+1:

**Review by jeremyg-lunarg — APPROVED** [review]
LGTM / +2

**Pull request merged**